### PR TITLE
V6 div only

### PR DESCRIPTION
**Pull request by emmahodcroft**
Should address **bug 1** of auspice issue [#745](https://github.com/nextstrain/auspice/issues/745).

We had left two 'TODO's to actually check that dates and mutations are present before adding `gt` and `num_date` as colorby values to the JSON. If those are there, Auspice trusts this, and will go looking for them (breaking error!)! 
(And as a side-note, `author` was getting auto-added too, though this doesn't seem to cause errors in Auspice.)

For `author` and `num_date`, this should have been taken care of by the function `_get_values`, since colorbys are only added `if _get_values(...`. 

However, rather than returning `None` if there are no values of this trait on the tree, what's returned is a set with one value, `None` (`{None}`). This means the answer to the 'if' above is True - and the two colorbys get added. 

I updated this so at the end of `_get_values` there's a check for length one and `None` value - if this is the case, it returns `None`, so the test is correct. There may be neater way of doing this.

....

However, this does not prevent 'Genotype'/`gt` from being added. If this remains, selecting it from the dropdown does no harm, but typing in any numbers does (though they aren't breaking):
image

To prevent this, we need to set variable `mutations_present` to False (at the moment, it's set always to True, with a #TODO tag).

I fixed this by creating function `check_muts` which counts the number of nodes with a `mutations` property. If 0, `None` is returned, and `mutations_present` can be set to False. 

This results in no colorbys being shown:
image

We may want Auspice to handle this in some special way? (Don't show the dropdown at all?) But at least this is correct and prevents users from doing things that aren't possible!

### CODE REVIEW

File: augur/export_v2.py [hunk 1]
```diff
@@ -603,13 +613,16 @@ def run_v2(args):
 
     add_metadata_to_tree(auspice_json["tree"], node_metadata)
 
+    mutations_present = False
+    if check_muts(node_metadata):
+        mutations_present = True
```
Comment (rneher) [inline augur/export_v2.py]: why not
`mutations_present = bool(check_muts(node_metadata))`

File: augur/export_v2.py [hunk 2]
```diff
@@ -103,6 +111,8 @@ def _get_values(key):
                 values_in_tree.add(node_properties.get(key, {}).get("value"))
             elif key in node_properties['traits']:
                 values_in_tree.add(node_properties['traits'][key]['value'])
+        if len(values_in_tree)==1 and None in values_in_tree:
+            return None
```
Comment (rneher) [inline augur/export_v2.py]: would be more consistent to return an empty set.

Comment (jameshadfield) [reply augur/export_v2.py]: Agreed -- this is a bug in (my) `values_in_tree` -- it shouldn't contain `None` (but may contain `False` if it's a Boolean variable)

File: augur/export_v2.py (hunk 1 above)
Comment (jameshadfield) [reply augur/export_v2.py]: and move it to line 625 😉

**Comment by jameshadfield** [comment]
I've just added some tests (to this branch, as they work on this one 😉 ) so our test builds now cover:
* div-only tree
* time-only tree
* BEAST parsing & export
* hidden-nodes
* zika (i.e. canonical FASTA build) -- v1 + v2
* TB (VCF build) -- v1 + v2
* TB-DRM v1 + v2

All can be triggered via `bash tests/builds/runner.sh` and `auspice develop tests/builds/auspice`.

I think this is going to be very helpful for development as we allow more diverse builds

**Comment by emmahodcroft** [comment]
> I've just added some tests (to this branch, as they work on this one 😉 ) so our test builds now cover:

This is a bit off-topic for this thread but I've just tried these and though they work in `augur`, I'm getting weird things (crashes) when I load in `auspice`. I should say I don't usually run via the `auspice view` command (but I wouldn't expect this to make any difference?). The errors are apparently minimised, so they aren't terribly helpful. 

When I put the test files in my `auspice/data` and view in running with `npm run dev` (my usual way), they seem fine. 

However, we have so many branches at the moment it might be worth waiting until some more are merged before I try to track this down to ensure it's not something silly.

**Comment by emmahodcroft** [comment]
Have incorporated the changes proposed by @rneher and @jameshadfield . There are now some conflicts - James I will leave these to you as you will be keeping these straight more than me - hope that's ok!

**Pull request merged**